app/controllers/foreman_deployments/api/v2/stacks_controller.rb

```diff
@@ -20,6 +20,18 @@ def create
           process_response @stack.save
         end
 
+        api :PUT, '/stacks/:id/', N_('Update imported stack')
+        param_group :stack, :as => :update
+        param :id, :identifier, :required => true
+        def update
+          if !@stack.configurations.empty? && !params[:stack][:definition].nil?
```

Comment:
Let's move this validation into the `Stack` model class. I can accept the code as is since it does what is intended, but I prefer moving logic to the lowest possible layer.
If we decide to leave it here, please file an issue to refactor this later.